src/angular/components/login.component.ts

```diff
@@ -37,7 +40,9 @@ export class LoginComponent implements OnInit {
 
     constructor(protected authService: AuthService, protected router: Router,
         protected platformUtilsService: PlatformUtilsService, protected i18nService: I18nService,
-        private storageService: StorageService, protected stateService: StorageService) { }
+        protected stateService: StorageService, protected environmentService: EnvironmentService,
```

Comment:
@vincentsalucci is this intended to be have two instances of a StorageService? Seems like it should be `stateService: StateService` ?

Reply:
I'm also wondering if the CryptoFunctionService needs to get added to the index.ts in the abstractions folder?

Reply:
( went ahead and threw together a PR for this stuff in case it is needed )